### DISCUSSION TITLE
AzuerAD registered application - how to limit its scope

### DISCUSSION THREAD

**Discussion by szymonslipek** [AzureCommunityPL/group#171] [Pomoc]
Czołem Azure people.

Case: Duży holding w jednym tenancie wiele firm córek.

Firmacórka1 - rejestruje aplikację w AzureAD z upranieniami do odczytu wszyskich maili w tenancie.
Firmacórka2 - "WTF?, nie zgadzamy się na to"
HoldingSecurityTeam  - Firmacórka2 ma rację, please kindly do the needfull.

Czy jest możliwość ograniczenia uprawnień zarejestrowanej aplikacji tylko do danej grupy userów?

W EXO rzecz prosta - ManagementScope (tylko domena X), dodanie serviceaccounta i określonej roli.... ale ma być bardziej nowocześnie - przez rejestracje apek.... 

Any hints?

**Comment by mgrabarz** [comment]
Moim zdaniem na poziomie rejestracji nie ma takiej opcji, ale w sumie nie ma to głębszego sensu.

W momencie gdy zarejestrujesz aplikację, domyślnie zinstancjonuje się ona w twoim AAD w postaci "Enterprise application". Na tym etapie już ma to sens, bo ograniczasz użycia tej aplikacji w kontekście Twojego AAD. Jak już ją znajdziesz w Enterprise applications poszukaj zakładki "Users and Groups" - tam ustaw wjazd do apki tylko grupie reprezentującej pracowników FirmyCórki1.

**Reply by szymonslipek** [reply]
Nie zrozumieliśmy się. Nie chodzi o kontrolę tego kto będzie używać aplikacji, tylko zakresu w jakim aplikacja będzie użuwać tenanta. Tj. w Enterprise apps jeśli ma uprawnienia do odczytywania zawartości skrzynek, to (wzorem ManagementScope/Role w EXO) jak ograniczyć tylko do np. wybranych domen a nie zostawiać pełny zakres.
Temat już rozwiązałem, za pomocą polis i grup. Dziękuję za zainteresowanie.

**Reply by mgrabarz** [reply]
W tym cały trik że piszesz o scopes, a nie app permissions. Scopes mają jedynie sens w kontekście uprawnień **delegowancyh** :)
Już tłumacze, dlaczego to ty mnie nie zrozumiałeś ;)
- Potrzebujesz ograniczyć by aplikacja miała dostęp do skrzynek dla userów w domenie podfirma1.com
- Rejestrujesz ją ze scopem do odczytu wszystkich skrzynek i wkładasz ograniczenie, że apka może być tylko używana przez userów z podfirma1.com
- W związku z tym że uprawnienie nie jest bezpośrednie, ale delegowane (bo scope), wymagane jest pozyskanie tokenów w Authorization Code  Grant w interakcji z userem.
- Tylko userzy z podfirma1.com mają autoryzację jej używania, co oznacza że apka nie dostanie tokenu dla userów z innych domen.
- W związku że to uprawnienie delegowane, nigdy nie uzyska dostępu do skrzynek userów z innych domen.

**Twoje rozwiązanie, bardziej wprost, oczywiście jest ok.**
